homeassistant/components/smarty/sensor.py

```diff
@@ -0,0 +1,237 @@
+"""
+Support for Salda Smarty XP/XV Ventilation Unit Sensors.
+
+For more details about this component, please refer to the documentation at:
+https://home-assistant.io/components/sensor.smarty/
+"""
+import logging
+
+from homeassistant.core import callback
+from homeassistant.components.smarty import (
+    DATA_SMARTY, SMARTY_NAME, SIGNAL_UPDATE_SMARTY)
+from homeassistant.const import (
+    TEMP_CELSIUS, DEVICE_CLASS_TEMPERATURE,
+    STATE_ON, STATE_OFF, STATE_UNKNOWN)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['smarty']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_CLASS_RPM = 'rpm'
+DEVICE_CLASS_ALARM = 'alarm'
+DEVICE_CLASS_WARNING = 'warning'
+DEVICE_CLASS_DAYS = 'day'
+DEVICE_CLASS_STATUS = 'status'
+RPM = 'rpm'
+DAYS = 'day'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the Smarty Sensor Platform."""
+    smarty = hass.data[DATA_SMARTY]
+    name = hass.data[SMARTY_NAME]
+
+    sensors = [SupplyAirTemperatureSensor(name, smarty),
```

Comment:
Instead of creating all of these classes that aren't materially different, I think you can combine it into a single class with an `if/elif/else` chain for the update logic and a data structure to map the name, class, and unit.

Reply:
It's easier to read without conditions.